metadata/src/test/java/org/apache/kafka/controller/ReplicationControlManagerTest.java

```diff
@@ -1922,6 +1922,135 @@ public void testReassignPartitions(short version) {
         assertEquals(NONE_REASSIGNING, replication.listPartitionReassignments(null, Long.MAX_VALUE));
     }
 
+    @ParameterizedTest
+    @ApiKeyVersionsSource(apiKey = ApiKeys.ALTER_PARTITION)
+    public void testAlterPartitionDisallowReplicationFactorChange(short version) {
```

Comment:
Maybe there already are such tests, in which case please point them out to me, but if not, could you add the same tests but with `setAllowReplicationFactorChange(true)`?

Reply:
Yes, we have `testReassignPartitions` and `testCancelReassignPartitions`. I believe these 2 tests already cover the scenario where altering the replication factor is possible when `allowReplicationFactorChange` is true (default value is true).